File: dotnet-core/Tea/Decorators/Elements/LogElement.cs
using System;
using OpenQA.Selenium;

namespace Tea
{
    public class LogElement : ElementDecorator
    {
        public LogElement(Element element):base(element) {}
        public override By By => Element?.By;

        public override string Text 
        {
            get
            {
                Console.WriteLine($"Element Text = {Element?.Text}");
                return Element?.Text;
            }
        }

        public override bool? Enabled 
        {
            get
            {
                Console.WriteLine($"Element enabled = {Element?.Enabled}");
                return Element?.Enabled;
            }
        }

        public override bool? Displayed
        {
            get
            {
                Console.WriteLine($"Element displayed = {Element?.Displayed}");
                return Element?.Displayed;
            }
        }
        public override void Click()
        {
            Console.WriteLine($"Element clicked: " + Element?.By.ToString());
            Element?.Click();
        }

        public override string GetAttribute(string attributeName)
        {
            Console.WriteLine($"Element's attribute = {Element?.GetAttribute(attributeName)}");
            return Element?.GetAttribute(attributeName);
        }

        public override void TypeText(string text)
        {
            Console.WriteLine($"Type Text = {text}");
            Element?.TypeText(text);        }
    }
}

File: java/tea-framework/src/main/java/com/tea/framework/SeleniumWrapper/Elements/LogElement.java
package com.tea.framework.SeleniumWrapper.Elements;

import org.openqa.selenium.By;

public class LogElement extends ElementDecorator {
    public LogElement(Element element) {
        super(element);
    }

    public static void write(String message) {
        System.out.println(message);
    }

    public By getBy() {
        return element.by;
    }

    public String getext() {
        write(String.format("Element Text = {%s}", this.element.getText()));
        return this.element.getText();
    }

    public boolean isEnabled() {
        write(String.format("Element enabled = {%s}", this.element.isEnabled()));
        return this.element.isEnabled();
    }

    public boolean isDisplayed() {
        write(String.format("Element displayed = {%s}", this.element.isDisplayed()));
        return this.element.isDisplayed();
    }

    public void Click() {
        write(String.format("Element clicked: " + this.element.getBy().toString()));
        this.element.Click();
    }

    public String GetAttribute(String attributeName) {
        write(String.format("Element's attribute = {%s}", this.element.GetAttribute(attributeName)));
        return this.element.GetAttribute(attributeName);
    }

    public void TypeText(String text) {
        write(String.format("Type Text = {%s}", text));
        this.element.TypeText(text);
    }
}